### DISCUSSION TITLE
CompositeElasticityTensor for crystal plasticity

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#16638] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,

I am working on coupling phase field and crystal plasticity. I have used ComputeElasticityTensorCP for describing the elasticity tensors for different phases in the system. I would like to combine them as in normal tensor mechanics using CompositeElasticityTensor. 
Does crystal plasticity allow using CompositeElasticityTensor for its computations? Or should I create a custom material to achieve this functionality?
Thanks,
Bala

**Comment by ngrilli** [comment, marked as answer]
Dear Bala @Bala-1005
 
The specific feature of ComputeElasticityTensorCP is the usage of the Euler angle reader.
If you want to use FiniteStrainCrystalPlasticity, you will always need those Euler angles.
You can start from another object to calculate the elasticity tensor, like CompositeElasticityTensor, and create your custom object in which
you add the Euler angle reader part of ComputeElasticityTensorCP.
I did something similar here:
https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/ComputeElasticityTensorMelting.C

Hope this helps,
Best Regards,
Nicolò Grilli
National University of Singapore

**Reply by Bala-1005** [reply]
Thank you Nicolo @ngrilli .I will take a look at your code and see if I can come up with something similar for compositeelasticity.

Bala
